src/containers/paint-editor.jsx

```diff
@@ -170,7 +171,8 @@ class PaintEditor extends React.Component {
                 log.error(`Mode not handled: ${this.props.mode}`);
                 this.props.changeMode(Modes.BRUSH);
             }
-        } else if (isBitmap(newFormat)) {
+        } else if (isBitmap(newFormat) &&
+                (this.props.mode in Modes && !(this.props.mode in BitmapModes))) {
```

Comment:
Why does this line of code not mirror the first `if` check above? E.g. above we only checked `if (isVector(newFormat) && (this.props.mode in BitmapModes))` as opposed to also adding `&& !(this.props.mode in Modes)` to that check as well...

I also wonder if we can consolidate these checks into helper functions (though not sure if there's a good way to do that... it seems like this is probably a good place for enums if `this.props.mode` can only ever be either `in BitmapModes` or `in Modes` but not both.

Reply:
It's actually that Modes is a superset of BitmapModes. Maybe it would be good for understandability to make a VectorModes enum

Reply:
I rewrote it adding VectorModes so hopefully it reads better now

Reply:
> It's actually that Modes is a superset of BitmapModes. Maybe it would be good for understandability to make a VectorModes enum

Ohhhh gotcha! I thought that Modes was vector modes. That makes sense now. Thank you for rewriting! I think the code is much clearer now! =]